expenseTracker/controllers/authController.js

```diff
@@ -9,19 +9,18 @@ const register = async (req, res) => {
   if (!email || !password || !username) {
     return res.status(400).json({ error: "Please provide all fields " });
   }
-
   try {
     // Check if user already exists
-    const existingUser = await User.findOne({ email });
+    const existingUser = await User.findOne({ 
+      $or: [{ email }, { username }]
```

Comment:
Nice usage of this Mongo feature!

You could also rely on `unique` constraints set on these fields in your `User` model, and then handle the response from Mongo after the fact. I like the clarity in doing it this way, though!